### PR TITLE
Updating disclaimer UX

### PR DESCRIPTION
**Pull request by gislawill**
This updates the splash screen and disclaimer. 
- There are subtle changes to the display of the splash screen buttons
- The disclaimer now fades in and out on a timer 
  - There's an x button to close it faster
-  The navigation controls stick around a bit longer

See linked videos:
| Before | After |
| - | - |
| https://github.com/CoastalResilienceExplorer/GlobalFloodExplorer/assets/8203830/5d41221a-6e05-49bc-809e-e1cdc1c7e34c | https://github.com/CoastalResilienceExplorer/GlobalFloodExplorer/assets/8203830/8c757cad-0503-46df-b258-16ef02c0efec |

### CODE REVIEW

File: src/Map.js [hunk 1]
```diff
@@ -118,12 +112,30 @@ export default function Map() {
...
   const [splashScreen, setSplashScreen] = useState(true);
   const [disclaimer, setDisclaimer] = useState(null);
+  const [navigationControls, setNavigationControls] = useState(null);
   const isTouch = window.matchMedia("(pointer: coarse)").matches;
 
+  useEffect(() => {
+    if (disclaimer) {
+      setTimeout(() => {
+        setDisclaimer(false);
+      }, 5000);
```
Comment (gislawill) [inline src/Map.js:122]: Leave disclaimer open for 5 seconds

File: src/Map.js [hunk 2]
```diff
@@ -118,12 +112,30 @@ export default function Map() {
...
+    }
+  }, [disclaimer]);
+
+  useEffect(() => {
+    if (navigationControls) {
+      setTimeout(() => {
+        setNavigationControls(false);
```
Comment (gislawill) [inline src/Map.js:129]: Leave navigationControls open for 10 seconds

File: src/components/fade-in-out.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import "./fade-in-out.css";
+
+interface FadeInOutProps {
+  children: ReactNode;
+  show: boolean;
+  additionalClasses?: string;
+  animationDuration?: number;
+}
+
+const FadeInOut: React.FC<FadeInOutProps> = ({
```
Comment (gislawill) [inline src/components/fade-in-out.tsx:11]: This allows components to fade in and out. The timing is controlled by animationDuration and the process is triggered by show

File: src/splash-screens/disclaimer-screen.tsx [hunk 4]
```diff
@@ -0,0 +1,161 @@
+import * as React from "react";
```
Comment (gislawill) [inline src/splash-screens/disclaimer-screen.tsx:1]: Migrated to typescript

File: src/splash-screens/disclaimer-screen.tsx [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+            {showClose && <AnimatedXIcon animationDuration={5000} />}
+          </div>
+          <div className="disclaimer-title">Disclaimer:</div>
+          <div className="disclaimer-body">
+            This map was created by the Coastal Resilience Lab to advance the
+            knowledge of mangroves for coastal flood protection. It is based off
+            a global model, and may contain issues when viewed at local scales.
+          </div>
+          <br />
+          {!isChrome && (
```
Comment (gislawill) [inline src/splash-screens/disclaimer-screen.tsx:100]: Only show this recommendation if not chrome

File: src/splash-screens/splash-screen.tsx [hunk 6]
```diff
@@ -1,5 +1,5 @@
 import * as React from "react";
```
Comment (gislawill) [inline src/splash-screens/splash-screen.tsx:1]: Migrated to typescript

File: src/types/declaration.d.ts [hunk 7]
```diff
@@ -0,0 +1,3 @@
+declare module "*.jpg";
+declare module "*.png";
+declare module "*.svg";
```
Comment (gislawill) [inline src/types/declaration.d.ts:3]: This tells typescript to not type check imported images (because it can't)

**Comment by lowriech** [comment]
This is awesome, I love the feel!  

Two notes:
- When trying to run with `bash local_spinup.sh`, I get some Typescript errors, asking me to run `npm i --save-dev @types/react`.  Since it was already a dev dependency, I ran `npm i @types/react` and it went away.  So, a little confused about the error, but can you check to make sure the Docker workflow is supported?
- Love the fadeout of the disclaimer, and the additional time of the controls.  What do you think about collapsing the controls into a clickable element after the timeout expires?  (That's the kind of thing Mike will generally enjoy)

**Pull request merged**